src/core/server/logging/appenders/rolling_file/policies/size_limit/size_limit_policy.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { schema, ByteSizeValue } from '@kbn/config-schema';
+import { LogRecord } from '@kbn/logging';
+import { RollingFileContext } from '../../rolling_file_context';
+import { TriggeringPolicy } from '../policy';
+
+export interface SizeLimitTriggeringPolicyConfig {
+  kind: 'size-limit';
+
+  /**
+   * The minimum size the file must have to roll over.
+   */
+  size: ByteSizeValue;
+}
+
+export const sizeLimitTriggeringPolicyConfigSchema = schema.object({
+  kind: schema.literal('size-limit'),
+  size: schema.byteSize({ min: '1b', defaultValue: '100mb' }),
```

Comment:
I haven't. @Kushmaro maybe you can help us here?

Reply:
By our [log delivery feature design](https://docs.google.com/document/d/1mbZ5zKlX9-6OSn4DmTX_RthUg_na4LFUsVZ2-cQPz0k/edit?usp=sharing#heading=h.lwfghv6kn3o4) (see storage section) we concluded that the smallest kibana instance in cloud will have around 1GB of storage space

is the 100mb referring to the log size rotation limit? cloud also enforces a quota on the log path, and I * think * that quota is 100mb total, so maybe a more sensible rotation limit would be 10mb.

@s-nel for awareness and further guidelines (he's on PTO though so may take a while)

Reply:
> is the 100mb referring to the log size rotation limit

That's the maximum size of the log file before it got rolled out.

Reply:
We use 100MB rollover size for ES audit logs and keep 2 files. We might choose something smaller for Kibana since the instances don't have much storage.